File: common/common-events/src/main/java/com/coltrack/events/analytics/AnalyticsEvent.java
package com.coltrack.events.analytics;

import java.math.BigDecimal;
import java.time.OffsetDateTime;
import java.util.Map;
import java.util.UUID;

public record AnalyticsEvent(
        UUID eventId,
        Integer schemaVersion,
        String eventType,
        String cameraId,

        Long trackId,
        String objectType,
        BigDecimal confidence,

        Long frameNumber,
        BigDecimal videoTimeSeconds,
        OffsetDateTime occurredAt,

        Map<String, Object> attributes
) {
}

File: frontend/src/types/AnalyticsEvent.ts
export interface AnalyticsEvent {
    eventId: string;
    schemaVersion: number;
    eventType: string;
    cameraId: string;
    trackId: number | null;
    objectType: string | null;
    confidence: number | null;
    frameNumber: number | null;
    videoTimeSeconds: number | null;
    occurredAt: string;
    receivedAt: string;
    recordingId: string | null;
    snapshotUrl: string | null;
    clipUrl: string | null;
    attributes: Record<string, unknown>;
}

export interface AnalyticsEventFilters {
    cameraId?: string;
    eventType?: string;
    objectType?: string;
    from?: string;
    to?: string;
    page?: number;
    size?: number;
    sort?: string;
}

export interface AnalyticsEventsPage {
    content: AnalyticsEvent[];
    number: number;
    size: number;
    totalElements: number;
    totalPages: number;
}
